### PR TITLE
Upgrade to Java 17

### LINKED ISSUE #2508: Java 17
### Problem

Java 17 will be release on 9/14/2021 and will be the first LTS release since Java 11. We should update our version of Java to take advantage of the new language features and optimizations in subsequent releases.

### Solution

- Update Java version in workflows, pom.xml, Dockerfiles, etc.
- Upgrade any dependent libraries
- Verify tests still pass
- Run docker images locally
- Verify no warnings at runtime

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by matheus-dallrosa**
**Description**:
Updates the java version to 17.0.1_12. The version was updated in all workflows, dockerfiles, and pom.xml files.

Fixes #2508 

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by xin-hedera** [comment]
should we use the latest temurin release 17.0.3? Or we need to align with ops?

**Comment by matheus-dallrosa** [comment]
> 17.0.3

Makes sense to use the latest. When I was searching for new versions of the docker image I only found the 17.0.1 ( had to look into the issues to find it ).

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
Still need to:
* Fix `hedera-mirror-rosetta/build/Dockerfile`
* Update `docs/installation.md` references to 11
* Add `actions/setup-java` to charts install, rest-api, and rosetta-api workflows that use `mvnw`

File: hedera-mirror-web3/pom.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
     </parent>
 
     <properties>
-        <java.version>11</java.version>
+        <java.version>17</java.version>
```
[the remark this answers is not in the document]
Comment (matheus-dallrosa) [reply hedera-mirror-web3/pom.xml]: Done.

File: .github/workflows/auto-release.yml [hunk 2]
```diff
@@ -83,7 +83,7 @@ jobs:
         with:
           cache: 'maven'
           distribution: 'temurin'
-          java-version: 11
+          java-version: 17.0.1
```
[the remark this answers is not in the document]
Comment (matheus-dallrosa) [reply .github/workflows/auto-release.yml]: Done.

File: hedera-mirror-web3/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (steven-sheehy) [reply hedera-mirror-web3/pom.xml]: Better to remove completely instead of duplicate it.

[the remark this answers is not in the document]
Comment (matheus-dallrosa) [reply hedera-mirror-web3/pom.xml]: Removed.

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**